src/main/java/org/folio/dao/UploadDefinitionDaoImpl.java

```diff
@@ -25,13 +26,116 @@
   private final Logger logger = LoggerFactory.getLogger(UploadDefinitionDaoImpl.class);
 
   private PostgresClient pgClient;
+  private String schema;
 
   public UploadDefinitionDaoImpl(Vertx vertx, String tenantId) {
     pgClient = PostgresClient.getInstance(vertx, tenantId);
+    this.schema = PostgresClient.convertToPsqlStandard(tenantId);
+  }
+
+  /**
+   * Functional interface for change UploadDefinition in blocking update statement
+   */
+  @FunctionalInterface
+  public interface UploadDefinitionMutator {
+    /**
+     * @param definition - Loaded from DB UploadDefinition
+     * @return - changed Upload Definition ready for save into database
+     */
+    Future<UploadDefinition> mutate(UploadDefinition definition);
+  }
+
+  public Future<UploadDefinition> updateBlocking(String uploadDefinitionId, UploadDefinitionMutator mutator) {
```

Comment:
plz, take a look on this update method

Reply:
Maybe updateExclusive ? , coz this is exclusive mode, which means the second select cannot proceed until the first one has completed or rolled back.